websocket.go

```diff
@@ -621,18 +620,19 @@ func (c *Client) SendProtobufParallelWithDecoder(nodes []*network.ServerIdentity
 				}
 			}
 
-			finish.Do(func() {
-				close(closed)
-			})
+			close(closed)
 
 			return <-siChan, nil
 		case err := <-errChan:
 			if opt.Quit() {
+				close(closed)
 				return nil, err
 			}
 			errs = append(errs, err)
 		}
 	}
+
+	close(closed)
```

Comment:
I don't understand how you can close a closed channel again without raising a panic - is this code-path tested?

Reply:
there is a close after each return ...

Reply:
Yes, and I put it in a `sync.Once` because it gave me panics otherwise. So I'm surprised that you can close the channel more than once...

Reply:
Can you have a look if the following commit might have introduced the error you're trying to fix here?

https://github.com/dedis/onet/commit/c575e990469fa077aa77a406612f8ed694039bf3#diff-8990d2c9c7f1690e60641d425c8932f7L600

It seems to me that it's more correct to close the channel in the go-routine than in the consumer for-loop, as the consumer might have to wait to be able to close the channel. As soon as a correct message is received, the channel should be closed. Then you won't have problems with the channel overflowing, and you can keep it with `parallel` length.